besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -2122,11 +2125,19 @@ private MiningParameters getMiningParameters() {
     if (miningParameters == null) {
       miningOptions.setGenesisBlockPeriodSeconds(
           getGenesisBlockPeriodSeconds(getActualGenesisConfigOptions()));
+      miningOptions.setMiningParametersMetricsSystem(getMiningParametersMetrics());
```

Comment:
just as an example on how you can simplify the PR, avoiding to move around the `MiningParameterMetrics` object, is to create a gauge like this:

```
      metricsSystem.get().createGauge(BesuMetricCategory.ETHEREUM,
              MIN_GAS_PRICE_GAUGE,
              "Gauge to measure the runtime value of min-priority-fee",
              () -> miningParameters.getMinTransactionGasPrice().toBigInteger().doubleValue());
```

since you can directly use the `miningParameters` object as source for the value, without duplicating the values in another class `MiningParameterMetrics` and having to change all the files to pass it around

Reply:
Thank you for the feedback. This makes complete sense. While understanding the other code sections, i thought of this but was not sure if it worked that way or not. Thank you for clarifying this. 

If my understanding is correct, this way i would not have to take miningParameterMetrics to the JSONRPC methods right and could just use the miningParameters to get the value. 
Please correct me if I am wrong.
Thank you

Reply:
Will work on the feedback and push changes as soon as possible. Thank you

Reply:
Updated the PR with feedback. would be grateful to get a review. Thank you